components/tyk-enterprise-portal/templates/statefulset-enterprise-portal.yaml

```diff
@@ -0,0 +1,235 @@
+        checksum/secrets: "{{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}"
+        {{- if .Values.mounts }}
+        {{- range $secret := .Values.mounts }}
+        "checksum/secret-{{ $secret.name }}": {{ include (print "mounts/%s" $secret.filename) . | sha256sum }}
+        {{- end }}
+        {{- end }}
+    spec:
+{{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        name: enterprise-portal-{{ .Chart.Name }}
+        securityContext:
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          privileged: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+              - all
+        env:
+          - name: PORTAL_LICENSEKEY
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+
+                key: EnterprisePortalLicense
+          - name: PORTAL_HOST_PORT
+            value: "{{ .Values.containerPort }}"
+          - name: ADMIN_EMAIL
+            value: "{{ .Values.global.adminUser.email }}"
+          - name: ADMIN_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+                key: EnterprisePortalAdminPassword
+          - name: PORTAL_DATABASE_DIALECT
+            value: "{{ .Values.storage.database.type }}"
+          - name: PORTAL_DATABASE_CONNECTIONSTRING
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+                key: EnterprisePortalStorgeConnectionString
+          - name: PORTAL_THEMING_THEME
+            value: "default"
+          - name: PORTAL_THEMING_PATH
+            value: "./themes"
+          - name: PROVIDER_NAME
+            value: "tyk"
+          - name: TYK_URL
+            value: "http{{ if .Values.global.tls.enterprisePortal }}s{{ end }}://dashboard-svc-{{ .Release.Name }}.{{ .Release.Namespace }}.svc:{{ .Values.global.servicePorts.dashboard }}"
```

Comment:
Shouldn't it be 
```suggestion
            value: "http{{ if .Values.global.tls.dashboard }}s{{ end }}://dashboard-svc-{{ .Release.Name }}.{{ .Release.Namespace }}.svc:{{ .Values.global.servicePorts.dashboard }}"
```

Reply:
Yes, agree. Also, this TYK_URL value seems missing to me. When we install `tyk-single-dc` chart, the dashboard svc has the following name:
```
dashboard-svc-tyk-single-dc-tyk-dashboard.tyk.svc
```
However, this TYK_URL has no `tyk-dashboard` suffix, which is the name of the dashboard subchart.

Reply:
Fixed